packages/client/src/BrubeckNode.ts

```diff
@@ -201,39 +203,44 @@ export default class BrubeckNode implements Context {
         try {
             this.destroySignal.assertNotDestroyed(this)
 
-            if (!this.cachedNode || !this.startNodeComplete) {
+            if (this.isStarting()) {
                 // use .then instead of async/await so
                 // this.cachedNode.publish call can be sync
                 return this.startNodeTask().then((node) => {
                     return node.publish(streamMessage)
                 })
             }
 
-            return this.cachedNode.publish(streamMessage)
+            return this.cachedNode!.publish(streamMessage)
         } finally {
             this.debug('publishToNode << %o', streamMessage.getMessageID())
         }
     }
 
-    async openPublishProxyConnectionOnStreamPart(streamPartId: StreamPartID, nodeId: string): Promise<void> {
+    async openProxyConnection(streamPartId: StreamPartID, nodeId: string, direction: ProxyDirection): Promise<void> {
```

Comment:
Could annotate this method with @internal as end-user should use this via `ProxyPublishSubscribe`. Also `closeProxyConnection`.